geode-protobuf-client/src/main/java/org/apache/geode/internal/cache/client/protobuf/Cache.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache.client.protobuf;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.internal.protocol.protobuf.v1.BasicTypes;
+import org.apache.geode.internal.protocol.protobuf.v1.ClientProtocol;
+import org.apache.geode.internal.protocol.protobuf.v1.ConnectionAPI;
+import org.apache.geode.internal.protocol.protobuf.v1.RegionAPI;
+
+@Experimental
+public class Cache {
+  final Socket socket;
+
+  Cache(String host, int port) throws Exception {
+    socket = new Socket(host, port);
+
+    final OutputStream outputStream = socket.getOutputStream();
+    outputStream.write(0x6E); // Magic byte
+    outputStream.write(0x01); // Major version
+
+    final InputStream inputStream = socket.getInputStream();
+    ClientProtocol.Message.newBuilder()
```

Comment:
Also broke this message and response.  You can see the new messages in handshake.proto.  The format of the messages is about the same, they're just renamed and now exist outside of the Message object.